Java/GameCenterAuthenticationHelper.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.security.Signature;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.Base64;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * {@link GameCenterAuthenticationHelper} implements server-side identity verification @see
+ * <a href="https://developer.apple.com/reference/gamekit/gklocalplayer/1515407-generateidentityverificationsign?language=objc">https://developer.apple.com/reference/gamekit/gklocalplayer/1515407-generateidentityverificationsign?language=objc</a>
+ *
+ * @author Johno Crawford (johno@hellface.com)
+ */
+public class GameCenterAuthenticationHelper {
+
+    private static final Logger logger = LogManager.getLogger(GameCenterAuthenticationHelper.class);
+
+    private static final long REFRESH_PERIOD = TimeUnit.MINUTES.toMillis(30);
+
+    private final ConcurrentMap<String, Certificate> refreshingCache = new ConcurrentHashMap<>();
+
+    private Timer refreshTimer;
+
+    @PostConstruct
+    public void initialize() {
+        try {
+            String publicKeyUrl = "https://static.gc.apple.com/public-key/gc-prod-2.cer";
```

Comment:
@normanmaurer hope you don't mind me reaching out but do you have any information on how caching publicKeyUrl should be done? ie. If Apple renews certificates will the number increase? gc-prod-3.cer and so on or is one expected to continually poll each url.